### DISCUSSION TITLE
VGGish mel-band frame length/hop size?

### DISCUSSION THREAD

**Discussion by seunggookim** [MTG/essentia#1251] [Essentia Models] [answered]
Hi!
I'm trying to use extracted embeddings from the VGGish model following the [usage example](https://essentia.upf.edu/models.html#audioset-vggish). I was able to get Numpy Array [# patches x 128]. In the description, the default `patchHopSize` = 93 frames and the default `patchSize` = 96 frames. I was wondering how long a Mel band frame was for this module. Given the audio file length I fed into the module, it appears that one frame is 10.0332 ms (160.5312 samples at 16 KHz) but I cannot figure out the number of samples in exact integers. Or perhaps the frame length was 25 ms (400 samples) and the hop size was 10 ms (160 samples)? Could you let me know how many audio samples (at 16 kHz) were used to create one frame? Or let me know where I can find the default values are set?
Perhaps returning the center time of each patch would be also convenient for many.

**Comment by palonso** [comment, marked as answer]
Hi @seunggookim,
Yes! The frame size is 400 samples (25ms) and the hop is 160 (10ms). Those parameters are inherited from the [original implementation](https://github.com/tensorflow/models/blob/master/research/audioset/vggish/vggish_params.py) and and we [hardcode](https://github.com/MTG/essentia/blob/2229b1839fa55b7900f2d74f120b1aac8b04dd62/src/algorithms/machinelearning/tensorflowpredictvggish.cpp#L98) them so that it is not possible to feed the model with incorrect mel-spectrograms.
Maybe you couldn't figure the exact numbers because of the zero-padding (`startFromZero=False`) introduced by the [frameCutter](https://essentia.upf.edu/reference/std_FrameCutter.html) algorithm?

I really like the idea of returning timestamps. Pleese feel free to fill a [feature request](https://github.com/MTG/essentia/issues/new) for it:)
